src/app/shared/components/map/map.component.ts

```diff
@@ -1,28 +1,47 @@
-import { Component, AfterViewInit, Input, Output, EventEmitter } from '@angular/core';
+import { Component, AfterViewInit, Input, Output, EventEmitter, OnDestroy } from '@angular/core';
 import * as Layer from 'leaflet';
 import { FormGroup } from '@angular/forms';
 import { GeolocationService } from 'src/app/shared/services/geolocation/geolocation.service';
 import { Coords } from '../../models/coords.model';
 import { Address } from '../../models/address.model';
 import { Workshop, WorkshopCard } from '../../models/workshop.model';
+import { Select } from '@ngxs/store';
+import { FilterState } from '../../store/filter.state';
+import { Observable } from 'rxjs';
+import { City } from '../../models/city.model';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
 
 @Component({
   selector: 'app-map',
   templateUrl: './map.component.html',
   styleUrls: ['./map.component.scss']
 })
-export class MapComponent implements AfterViewInit{
+export class MapComponent implements AfterViewInit, OnDestroy{
 
-  defaultKievCoords: Coords = { lat: 50.462235, lng: 30.545131 };
-  zoom: number = 11;
+  @Select(FilterState.city)
+  city$ :Observable<City>;
+
+  destroy$: Subject<boolean> = new Subject<boolean>();
+
+  public defaultCoords: Coords;
+  public zoom: number = 11;
 
   @Input() addressFormGroup: FormGroup;
   @Input() workshops: WorkshopCard[];
 
   @Output() setAddressEvent = new EventEmitter<Address>();
   @Output() selectedAddress = new EventEmitter<Address>();
 
-  constructor(private geolocationService: GeolocationService) { }
+  constructor(private geolocationService: GeolocationService) { 
+    this.city$
+    .pipe(takeUntil(this.destroy$))
+    .subscribe((city)=> {
+      const coords = {lat:city?.latitude, lng:city?.longitude};
```

Comment:
I believe it is better to filter city in the pipe instead of adding check in the declaration